socialsharing.php

```diff
@@ -1,277 +1,375 @@
+                    true).'&conf=6&configure='.$this->name.'&tab_module='.$this->tab.'&module_name='.$this->name);
+
+        }
+
+        $helper = new HelperForm();
+        $helper->submit_action = 'submitSocialSharing';
+        $helper->currentIndex = $this->context->link->getAdminLink('AdminModules',
+                false).'&configure='.$this->name.'&tab_module='.$this->tab.'&module_name='.$this->name;
+        $helper->token = Tools::getAdminTokenLite('AdminModules');
+        $helper->tpl_vars = array('fields_value' => $this->getConfigFieldsValues());
+
+        $fields = array();
+        foreach (self::$networks as $network) {
+            $fields[] = array(
+                'type' => 'switch',
+                'label' => $network,
+                'name' => 'PS_SC_'.Tools::strtoupper($network),
+                'values' => array(
+                    array(
+                        'id' => Tools::strtolower($network).'_active_on',
+                        'value' => 1,
+                        'label' => $this->l('Enabled')
+                    ),
+                    array(
+                        'id' => Tools::strtolower($network).'_active_off',
+                        'value' => 0,
+                        'label' => $this->l('Disabled')
+                    )
+                )
+            );
+        }
+
+        return $this->html.$helper->generateForm(array(
+            array(
+                'form' => array(
+                    'legend' => array(
+                        'title' => $this->displayName,
+                        'icon' => 'icon-share'
+                    ),
+                    'input' => $fields,
+                    'submit' => array(
+                        'title' => $this->l('Save')
+                    )
+                )
+            )
+        ));
+    }
+
+    /**
+     * Hook between head tags
+     *
+     * @param array $params Hook parameters
+     * @return string HTML to be displayed
+     */
+    public function hookDisplayHeader($params)
+    {
+        if (!isset($this->context->controller->php_self) || !in_array($this->context->controller->php_self,
+                array('product', 'products-comparison'))
+        ) {
+            return '';
+        }
+
+        $this->context->controller->addCss($this->_path.'css/socialsharing.css');
+        $this->context->controller->addJS($this->_path.'js/socialsharing.js');
+
+        if ($this->context->controller->php_self == 'product') {
+            $product = $this->context->controller->getProduct();
+
+            if (!Validate::isLoadedObject($product)) {
+                return '';
+            }
+            if (!$this->isCached('socialsharing_header.tpl',
+                $this->getCacheId('socialsharing_he ader|'.
+                    (isset($product->id) && $product->id ? (int)$product->id : '')))
+            ) {
+                if (!defined('_PS_PRICE_COMPUTE_PRECISION_')) {
```

Comment:
Yep, release early, release often, totally agree! But don't forget that these modules are also pushed to older versions.

I'm afraid we also can't ignore the the devs who are forced to make their modules compatible with older versions :)
This is the first thing I see (with `xdebug`) whenever I fire up a new instance of the first stable 1.6 version:
![1605socialsharin](https://cloud.githubusercontent.com/assets/6775736/14590399/b3e7a48e-04f9-11e6-9ca7-cd698f96e239.png)

Would be worth spilling a few CPU cycles on newer PS versions, right?